controller.go

```diff
@@ -444,6 +456,10 @@ func main() {
 	if err != nil {
 		log.Fatal(err)
 	}
+	if err = kubeAdapter.NewInclusterConfigClientset(ctx); err != nil {
+		log.Fatal(err)
+	}
+	kubeAdapter.WithTargetCNIPodSelector(targetCNIPodSelector)
```

Comment:
We should only do this in case `awsAdapter.TargetCNI.Enabled` or do I miss something?

Reply:
Logically yes, it is used only when enabled. Since we have the default flag `aws.DefaultTargetCNILabelSelector` always available it can parse any time and we don't need to care for nil config.
But if you prefer I can move that behind a case?

Reply:
yes please

Reply:
done